docs/examples/plot_2D_basis_function.py

```diff
@@ -0,0 +1,154 @@
+# -*- coding: utf-8 -*-
+
+"""
+# Plotting 2D Basis Functions
+
+## Defining a 2D Basis Function
+
+To model the response of a neuron to the position of an animal in an arena, we need a 2D model for stimuli.
+In this case, the stimuli are the 2D coordinates (x, y) representing the animal's position at each time step.
+
+We can model a response function for multi-dimensional stimuli as the external product of two 1D basis functions.
+The response function approximation for the 2D case would be:
+
+$$
+f(x, y) \\approx \sum_{ij} \\alpha_{ij} \, a_i (x) b_j(y).
+$$
+
+Here, we define this 2D basis function as the multiplication of two 1D basis objects.
+"""
+
+import numpy as np
+import matplotlib.pyplot as plt
+import neurostatslib as nsl
+
+# Define 1D basis objects
+a_basis = nsl.basis.MSplineBasis(n_basis_funcs=20, order=3)
+b_basis = nsl.basis.MSplineBasis(n_basis_funcs=10, order=2)
+
+# 2D basis function as the product of the two 1D basis objects
+prod_basis = a_basis * b_basis
```

Comment:
Since the later header says "Additive basis", the one for this section should say "Multiplicative basis" (or similar)

Also, should explain why someone would do this. E.g., "A multiplicative basis is an efficient way of generating a simple representation of x,y position in physical or visual space. etc"